From: Didier Roche <didrocks@ubuntu.com>
Date: Wed, 18 Mar 2015 17:11:00 +0100
Subject: PrivateTmp shouldn't require tmpfs

As PrivateTmp is requiring tmp.mount, this one will mount (but only after boot)
/tmp as tmpfs adding a Requires=tmp.mount to the unit. This change downgrades
the requirements to an after relationship.
---
 src/core/unit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/unit.c b/src/core/unit.c
index 565455b..93487d8 100644
--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -807,7 +807,7 @@ int unit_add_exec_dependencies(Unit *u, ExecContext *c) {
                 return 0;
 
         if (c->private_tmp) {
-                r = unit_require_mounts_for(u, "/tmp");
+                r = unit_add_dependency_by_name(u, UNIT_AFTER, "tmp.mount", NULL, true);
                 if (r < 0)
                         return r;
 
